d2compiler/compile.go

```diff
@@ -307,9 +307,28 @@ func (c *compiler) compileLabel(attrs *d2graph.Attributes, f d2ir.Node) {
 
 func (c *compiler) compileReserved(attrs *d2graph.Attributes, f *d2ir.Field) {
 	if f.Primary() == nil {
-		if f.Composite != nil && !strings.EqualFold(f.Name, "class") {
+		compositeAccepted := strings.EqualFold(f.Name, "class") || strings.EqualFold(f.Name, "constraint")
+		if f.Composite != nil && !compositeAccepted {
```

Comment:
composite might still be nil below.

the switch statement should go inside `if f.Composite != nil`, and then the default case is the error